src/components/editor/property/PropertyTemplateOutline.jsx

```diff
@@ -82,49 +56,39 @@ export class PropertyTemplateOutline extends Component {
   }
 
   addPropertyTypeRows = (property) => {
-    let existingJsx; let
-      propertyJsx
-    const newOutput = this.state.propertyTypeRow
+    const newOutput = [...this.state.propertyTypeRow]
+    let propertyJsx
 
     if (isResourceWithValueTemplateRef(property)) {
-      const isAddDisabled = !templateBoolean(property.repeatable)
-      const newReduxPath = Object.assign([], this.props.reduxPath)
+      const isAddDisabled = !templateBoolean(property.repeatable) || newOutput.length > 0
 
-      newReduxPath.push(property.propertyURI)
-
-      propertyJsx = <ResourceProperty propertyTemplate={property}
-                                      reduxPath={newReduxPath}
+      propertyJsx = <ResourceProperty key={shortid.generate()}
+                                      propertyTemplate={property}
+                                      reduxPath={this.props.reduxPath}
                                       nestedResourceTemplates={this.state.nestedResourceTemplates}
                                       handleAddClick={this.handleAddClick}
                                       addButtonDisabled={isAddDisabled} />
     } else {
-      propertyJsx = <PropertyComponent index={0} propertyTemplate={property} reduxPath={this.props.reduxPath} />
+      propertyJsx = <PropertyComponent key={shortid.generate()} propertyTemplate={property} reduxPath={this.props.reduxPath} />
     }
 
-    newOutput.forEach((propertyJsx) => {
-      if (this.props.propertyTemplate.propertyURI === propertyJsx.props.propertyTemplate.propertyURI) {
-        existingJsx = propertyJsx
-      }
-    })
-    if (existingJsx === undefined) {
-      newOutput.push(
-        <PropertyTypeRow
-          key={shortid.generate()}
-          handleAddClick={this.props.handleAddClick}
-          reduxPath={this.props.reduxPath}
-          addButtonDisabled={this.props.addButtonDisabled}
-          propertyTemplate={property}>
-          {propertyJsx}
-        </PropertyTypeRow>,
-      )
-    }
+    newOutput.push(
+      <PropertyTypeRow
+        key={shortid.generate()}
+        handleAddClick={this.props.handleAddClick}
+        reduxPath={this.props.reduxPath}
+        addButtonDisabled={this.props.addButtonDisabled}
+        propertyTemplate={property}>
+        {propertyJsx}
+      </PropertyTypeRow>,
+    )
 
     this.setState({ propertyTypeRow: newOutput, rowAdded: true })
   }
 
   render() {
     return (
-      <div className="rtOutline" key={shortid.generate()}>
+      <div className="rtOutline">
```

Comment:
So the key attribute that was here is now moved to PropertyComponent?

Reply:
Not that I fully understand this aspect of react, but this was forcing new components every time in rendered rather than re-using existing.